lib/include/jxl/types.h

```diff
@@ -109,6 +109,20 @@ typedef struct {
   size_t align;
 } JxlPixelFormat;
 
+/** Boolean option to enable, disable or set to default.
+ */
+typedef enum {
+  /** Leaves the option to the default, enabled or disabled.
+   */
+  JXL_OVERRIDE_DEFAULT = -1,
+  /** Force-disables the option.
+   */
+  JXL_OVERRIDE_FALSE = 0,
+  /** Force-enables the option.
+   */
+  JXL_OVERRIDE_TRUE = 1,
+} JxlOptionOverride;
```

Comment:
I would love to avoid having this "override" concept in the API.

Reply:
A very rigid alternative is a separate enum for each setting, but I am reluctant to make it that rigid because the encoder settings themselves are not very rigid: there are many encoder settings, they will change, they will get new levels/strengths added, new ones will be added, old ones will get deprecated.

Integer values and booleans can be ok.